### PR TITLE
Add gas tank refilling

### PR DESCRIPTION
**Pull request by AtomicFox556**
#### Summary
Content "Add gas tank refilling"

#### Purpose of change
Add ways to refill oxygen tanks and scuba tanks, making more use of machinery added in #55937 and expanding some more on it.

#### Describe the solution
Add gas pump appliance, and recipes to refill all the oxygen and scuba tanks. Had to rewrite how oxygen tanks and scuba tanks work by making them into magazines and adding separate items that make use of them, in order to make them refillable.

#### Describe alternatives you've considered
None.

#### Testing
Tried using scuba tank and oxygen tank in game to make sure they behave correctly.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
I've seen air compressors in dive shops that when examined indicate that they could be used for filling scuba tanks. However, I haven't verified that they actually work.

**Comment by AtomicFox556** [comment]
> I've seen air compressors in dive shops that when examined indicate that they could be used for filling scuba tanks. However, I haven't verified that they actually work. It might be useful to check that your work complements what's there rather than duplicate parts of it.

There is indeed an air compressor in existence already, but the difference here is that scuba tanks in most cases contain more oxygen than air does (starting from 32%, usually not higher than 40% and up to about 50% at most), so you can't use a normal air compressor to fill one up; you'd need to filter out some nitrogen first.

**Comment by kevingranade** [comment]
Synthesizing gasses on the spot was one thing for ammonia synthesis, but I don't think it's a normal practice elsewhere, and it's especially egregious for it to be the only option.
What we need is mixing existing gasses as the primary option, so you might find Oxygen and Nitrogen tanks, and set them up with a blending system to yield your diving mix.
Short of that, you absolutely can use compressed atmospheric gas, but it limits your safe dive depth, I would focus on that.

**Comment by AtomicFox556** [comment]
> Short of that, you absolutely can use compressed atmospheric gas, but it limits your safe dive depth, I would focus on that.

I'll stick to this method for refilling diving tanks then, considering that decompression sickness isn't a thing yet.

File: data/json/furniture_and_terrain/appliances.json [hunk 1]
```diff
@@ -605,6 +605,36 @@
...
+    "pseudo_tools": [ { "id": "gaspump" } ],
+    "breaks_into": [
+      { "item": "steel_lump", "count": [ 2, 4 ] },
+      { "item": "steel_chunk", "count": [ 2, 4 ] },
+      { "item": "scrap", "count": [ 2, 4 ] },
+      { "item": "hd_pipe", "count": [ 1, 2 ] },
+      { "item": "hd_pipe_fittings", "prob": 50 },
+      { "item": "hd_compressor", "prob": 25 },
+      { "item": "hd_valve", "prob": 25 },
+      { "item": "hd_valve", "prob": 25 },
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/appliances.json:633]: Is this a duplicate?

Comment (AtomicFox556) [reply data/json/furniture_and_terrain/appliances.json]: It is. Fixed now.

**Pull request closed without merging**